ytypes/bool_type.go

```diff
@@ -30,6 +32,14 @@ func validateBool(schema *yang.Entry, value interface{}) error {
 		return err
 	}
 
+	// If the schema is type empty, check that it had the correct type name.
```

Comment:
better to split this to a new function validateEmpty rather than overloading validateBool.

Reply:
Done.